### PR TITLE
Add FadeBy as an action

### PR DESCRIPTION
**Pull request by hawkwood**

### CODE REVIEW

File: cocos/2d/CCActionInterval.h [hunk 1]
```diff
@@ -834,6 +834,40 @@ class CC_DLL FadeTo : public ActionInterval
...
+	
+    /** initializes the action with duration and opacity */
+    bool initWithDuration(float duration, GLubyte opacity);
+	
+protected:
+    GLubyte _byOpacity;
+    GLubyte _toOpacity;
+    GLubyte _fromOpacity;
+    friend class FadeOut;
+    friend class FadeIn;
```
Comment (minggo) [inline cocos/2d/CCActionInterval.h]: Why should declare these classes as friend?

**Comment by minggo** [comment]
Could you add a test case?

**Comment by hawkwood** [comment]
They do not need to be, that was a carry over from copying FadeTo.

**Comment by minggo** [comment]
So could you please modify it and add a test case?
Thanks.

**Comment by hawkwood** [comment]
I made the change, but am not sure what/how to add a test case.  Do you want me to put a code snippet here showing how to use it, or something else?

**Comment by minggo** [comment]
You can add a test case in `tests/cpp-tests/Classes/ActionsTest`.

**Comment by minggo** [comment]
@hawkwood 
Any update? Do you have any question about test case?

**Comment by hawkwood** [comment]
Once I started to add a test case, I realized that my code had some significant issues.  I have fixed them and added the test case as requested.

**Comment by minggo** [comment]
Thanks, could you please fix indent? 
Replace tab with 4 spaces.

**Comment by minggo** [comment]
[ci rebuild]

**Comment by minggo** [comment]
[ci rebuild]

**Comment by minggo** [comment]
[ci rebuild]

File: cocos/2d/CCActionInterval.cpp [hunk 2]
```diff
@@ -1953,6 +1953,85 @@ void FadeTo::update(float time)
...
+    if (_target)
+    {
+        float dt = (time - _lastTime);
+        _lastTime = time;
+
+        GLubyte fromOpacity = _target->getOpacity();
+        GLubyte nextOpacity = fromOpacity;
+        if (_additive) {
+            float val = _byOpacity * dt + _lastFrac;
+            float num = roundf( val );
```
Comment (minggo) [inline cocos/2d/CCActionInterval.cpp:2016]: It will cause compiling error on windows.

Comment (minggo) [inline cocos/2d/CCActionInterval.cpp:2016]: `roundf` is not defined in vs2012.

**Comment by hawkwood** [comment]
Huh, looks like it wasn't added till vs2013.

**Comment by minggo** [comment]
Yep.
So i think we should work around it.

**Comment by minggo** [comment]
Any update?

**Pull request closed without merging**